packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/method_channel/method_channel_google_maps_flutter.dart

```diff
@@ -0,0 +1,468 @@
+    final List<dynamic> latLng = await channel(mapId)
+        .invokeMethod<List<dynamic>>(
+            'map#getLatLng', screenCoordinate.toJson());
+    return LatLng(latLng[0], latLng[1]);
+  }
+
+  /// Programmatically show the Info Window for a [Marker].
+  ///
+  /// The `markerId` must match one of the markers on the map.
+  /// An invalid `markerId` triggers an "Invalid markerId" error.
+  ///
+  /// * See also:
+  ///   * [hideMarkerInfoWindow] to hide the Info Window.
+  ///   * [isMarkerInfoWindowShown] to check if the Info Window is showing.
+  @override
+  Future<void> showMarkerInfoWindow(
+    MarkerId markerId, {
+    @required int mapId,
+  }) {
+    assert(markerId != null);
+    return channel(mapId).invokeMethod<void>(
+        'markers#showInfoWindow', <String, String>{'markerId': markerId.value});
+  }
+
+  /// Programmatically hide the Info Window for a [Marker].
+  ///
+  /// The `markerId` must match one of the markers on the map.
+  /// An invalid `markerId` triggers an "Invalid markerId" error.
+  ///
+  /// * See also:
+  ///   * [showMarkerInfoWindow] to show the Info Window.
+  ///   * [isMarkerInfoWindowShown] to check if the Info Window is showing.
+  @override
+  Future<void> hideMarkerInfoWindow(
+    MarkerId markerId, {
+    @required int mapId,
+  }) {
+    assert(markerId != null);
+    return channel(mapId).invokeMethod<void>(
+        'markers#hideInfoWindow', <String, String>{'markerId': markerId.value});
+  }
+
+  /// Returns `true` when the [InfoWindow] is showing, `false` otherwise.
+  ///
+  /// The `markerId` must match one of the markers on the map.
+  /// An invalid `markerId` triggers an "Invalid markerId" error.
+  ///
+  /// * See also:
+  ///   * [showMarkerInfoWindow] to show the Info Window.
+  ///   * [hideMarkerInfoWindow] to hide the Info Window.
+  @override
+  Future<bool> isMarkerInfoWindowShown(
+    MarkerId markerId, {
+    @required int mapId,
+  }) {
+    assert(markerId != null);
+    return channel(mapId).invokeMethod<bool>('markers#isInfoWindowShown',
+        <String, String>{'markerId': markerId.value});
+  }
+
+  /// Returns the current zoom level of the map
+  @override
+  Future<double> getZoomLevel({
+    @required int mapId,
+  }) {
+    return channel(mapId).invokeMethod<double>('map#getZoomLevel');
+  }
+
+  /// Returns the image bytes of the map
+  @override
+  Future<Uint8List> takeSnapshot({
+    @required int mapId,
+  }) {
+    return channel(mapId).invokeMethod<Uint8List>('map#takeSnapshot');
+  }
+
+  /// This method builds the appropriate platform view where the map
+  /// can be rendered.
+  /// The `mapId` is passed as a parameter from the framework on the
+  /// `onPlatformViewCreated` callback.
+  @override
+  Widget buildView(
+      Map<String, dynamic> creationParams,
+      Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers,
+      PlatformViewCreatedCallback onPlatformViewCreated) {
+    if (defaultTargetPlatform == TargetPlatform.android) {
```

Comment:
This is the MethodChannel implementation, which can be platform-specific (at least for the devices that use MethodChannel).

It needs to live in this package, so it can be the "default" implementation, if the plugin is not overriding anything.

(This code *is* platform-specific already, because it uses MethodChannels, pretty much)